server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -143,15 +120,29 @@ static void clearLoadedModules()
    */
   public synchronized static <T> Collection<T> getFromExtensions(ExtensionsConfig config, Class<T> clazz)
   {
-    final TeslaAether aether = getAetherClient(config);
     final Set<T> retVal = Sets.newHashSet();
-    final Set<String> extensionNames = Sets.newHashSet();
+    final Set<String> loadedExtensionNames = Sets.newHashSet();
 
-    for (String coordinate : config.getCoordinates()) {
-      log.info("Loading extension[%s] for class[%s]", coordinate, clazz.getName());
-      try {
-        URLClassLoader loader = getClassLoaderForCoordinates(aether, coordinate, config.getDefaultVersion());
+    if (config.searchCurrentClassloader()) {
+      for (T module : ServiceLoader.load(clazz, Initialization.class.getClassLoader())) {
```

Comment:
(should not block this PR) I was looking at this part of the code recently and wondering if this should be using the thread context class loader by default instead of the Initialization.class's classloader.